### PR TITLE
add missing l1 transaction receipt field

### LINKED ISSUE #575: L1HandlerTransactionReceipt is missing message_hash
See [here](https://github.com/starkware-libs/starknet-specs/blob/v0.7.1/api/starknet_api_openrpc.json#L3072)

### PR DESCRIPTION
**Pull request by rianhughes**
closes https://github.com/NethermindEth/starknet.go/issues/575

### CODE REVIEW

File: rpc/transaction_test.go [hunk 1]
```diff
@@ -270,8 +336,26 @@ func TestTransactionReceipt(t *testing.T) {
...
 		testConfig.provider.c = spy
 		txReceiptWithBlockInfo, err := testConfig.provider.TransactionReceipt(context.Background(), test.TxnHash)
-		require.Nil(t, err)
-		require.Equal(t, test.ExpectedResp, *txReceiptWithBlockInfo)
+		switch test.ExpectedTxnType {
+		case TransactionType_Invoke:
+			require.Nil(t, err)
+			rec, ok := (txReceiptWithBlockInfo.TransactionReceipt).(InvokeTransactionReceipt)
+			require.True(t, ok)
+			rec2, ok := (txReceiptWithBlockInfo.TransactionReceipt).(InvokeTransactionReceipt)
```
Comment (cicr99) [inline rpc/transaction_test.go]: Not sure I understand, is there a difference here between `rec` and `rec2`?

File: rpc/transaction_test.go [hunk 2]
```diff
@@ -270,8 +336,26 @@ func TestTransactionReceipt(t *testing.T) {
...
+			require.True(t, ok)
+			require.Equal(t, rec2, rec)
+		case TransactionType_L1Handler:
+			require.Nil(t, err)
+			rec, ok := (txReceiptWithBlockInfo.TransactionReceipt).(L1HandlerTransactionReceipt)
+			require.True(t, ok)
+			rec2, ok := (txReceiptWithBlockInfo.TransactionReceipt).(L1HandlerTransactionReceipt)
```
Comment (cicr99) [inline rpc/transaction_test.go]: Same question here

File: rpc/transaction_test.go (hunk 1 above)
Comment (rianhughes) [reply rpc/transaction_test.go]: Whoops :) Fixed

File: rpc/transaction_test.go (hunk 2 above)
Comment (rianhughes) [reply rpc/transaction_test.go]: Same as above

**Review by cicr99 — APPROVED** [review]
lgtm

**Pull request merged**